elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -252,56 +251,156 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
     }
 
     private void populateInheritance(ElideTableConfig elideTableConfig) {
-        for (Table table : elideTableConfig.getTables()) {
-            if (table.hasParent()) {
-                Map<String, Measure> measures = getInheritedMeasures(table, new HashMap<String, Measure>());
-                table.setMeasures(new ArrayList<Measure>(measures.values()));
+        //ensures validation is run before populate always.
+        validateInheritance(this.elideTableConfig);
 
-                Map<String, Dimension> dimensions = getInheritedDimensions(table, new HashMap<String, Dimension>());
-                table.setDimensions(new ArrayList<Dimension>(dimensions.values()));
+        Set<Table> processed = new HashSet<Table>();
+        elideTableConfig.getTables().stream().forEach(table -> {
+            populateInheritance(table, processed);
+        });
+    }
 
-                Map<String, Join> joins = getInheritedJoins(table, new HashMap<String, Join>());
-                table.setJoins(new ArrayList<Join>(joins.values()));
+    private void populateInheritance(Table table, Set<Table> processed) {
+        if (table.hasParent() && !processed.contains(table)) {
```

Comment:
Nitpick - but I would invert this if condition and return early if the opposite is true.  Then the entire function doesn't have to be nested inside this if block (it helps with line length and readability).